Resources/public/js/views/ez-trashview.js

```diff
@@ -21,9 +21,17 @@ YUI.add('ez-trashview', function (Y) {
      * @constructor
      * @extends eZ.TemplateBasedView
      */
-    Y.eZ.TrashView = Y.Base.create('trashView', Y.eZ.TemplateBasedView, [], {
+    Y.eZ.TrashView = Y.Base.create('trashView', Y.eZ.TemplateBasedView, [Y.eZ.SelectionTable], {
+        events: {
+            '.ez-trashitem-box': {
+                'change': '_itemSelectionChange'
+            },
+        },
+
         initializer: function () {
             this.on('*:minimizeTrashBarAction', this._handleMinimizeTrashBar);
+            this.on('*:restoreTrashItemsAction', this._restoreTrashItems);
+            this.after('activeChange', this._itemSelectionChange);
```

Comment:
weird :) to avoid that, the restore button action view should be disabled by default.